data/transactions/logic/box.go

```diff
@@ -207,29 +231,28 @@ func opBoxPut(cx *EvalContext) error {
 	value := string(cx.stack[last].Bytes)
 	name := string(cx.stack[prev].Bytes)
 
-	err := cx.availableBox(name, boxWrite, 0 /* unused for write */)
-	if err != nil {
-		return err
-	}
-	box, ok, err := cx.Ledger.GetBox(cx.appID, name)
+	contents, exists, err := cx.availableBox(name, boxWrite, uint64(len(value)))
 	if err != nil {
 		return err
 	}
 
 	cx.stack = cx.stack[:prev]
 
-	if ok {
-		/* the box exists, so the replacement must match size */
-		if len(box) != len(value) {
-			return fmt.Errorf("attempt to box_put wrong size %d != %d", len(box), len(value))
+	if exists {
+		/* the replacement must match existing size */
+		if len(contents) != len(value) {
+			return fmt.Errorf("attempt to box_put wrong size %d != %d", len(contents), len(value))
 		}
 		return cx.Ledger.SetBox(cx.appID, name, value)
 	}
+
 	/* The box did not exist, so create it. */
+	_, err = createCheck(cx, name, uint64(len(contents)))
```

Comment:
@jannotti I think there's a bug on the highlighted line.  I attempted to diagnose in https://github.com/jannotti/go-algorand/pull/15.  Can you have a look to see if you agree?

Reply:
@jannotti https://github.com/algorand/go-algorand/pull/4622/commits/01565f94d484bf46cb89a54b3d890bbdd43969a1 brings in a fix for the bug from the child PR.

Can you assess if you want to add these highlighted tests https://github.com/jannotti/go-algorand/pull/15/files#diff-4cce2c8cab90d57a3809d9df791fb702dd381a8614706a4eb98549002cc34c5eR77-R90?

Particularly since it feels low effort, I see value in exercising the highlighted cases for  `box_create` and `box_put` rather than only `box_create`.   Either way, I leave the final call up to you - Once there's a response, I'm good to resolve the thread.

Reply:
I think I've taken everything from jannotti#15 that wasn't covered by something taken previously, or a new test I added that tests all the box opcodes for 0 length names and >64 length names.